tests/__init__.py

```diff
@@ -24,6 +24,7 @@
 import pytest
 from contextlib import contextmanager
 from linkcheck import LinkCheckerInterrupt
+from builtins import str
```

Comment:
I'm not familiar with `future`.  Is this equivalent to `str = unicode` on Python 2?

I'm not sure I like that, it seems very subtle.  I much prefer the approach `six` took where you have a new `six.text_type` to mean `unicode` on Python 2 and `str` on Python 3.

(Still, this is test code, I'm not going to veto anything.)